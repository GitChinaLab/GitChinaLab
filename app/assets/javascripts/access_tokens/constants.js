// Token types
export const FEED_TOKEN = 'feedToken';
export const INCOMING_EMAIL_TOKEN = 'incomingEmailToken';
export const STATIC_OBJECT_TOKEN = 'staticObjectToken';
