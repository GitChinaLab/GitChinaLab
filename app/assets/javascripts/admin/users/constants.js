import { s__, __ } from '~/locale';

export const USER_AVATAR_SIZE = 32;

export const LENGTH_OF_USER_NOTE_TOOLTIP = 100;

export const I18N_USER_ACTIONS = {
  edit: __('Edit'),
  userAdministration: s__('AdminUsers|User administration'),
  unlock: __('Unlock'),
  block: s__('AdminUsers|Block'),
  unblock: s__('AdminUsers|Unblock'),
  approve: s__('AdminUsers|Approve'),
  reject: s__('AdminUsers|Reject'),
  deactivate: s__('AdminUsers|Deactivate'),
  activate: s__('AdminUsers|Activate'),
  ldapBlocked: s__('AdminUsers|Cannot unblock LDAP blocked users'),
  delete: s__('AdminUsers|Delete user'),
  deleteWithContributions: s__('AdminUsers|Delete user and contributions'),
  ban: s__('AdminUsers|Ban user'),
  unban: s__('AdminUsers|Unban user'),
};

export const CONFIRM_DELETE_BUTTON_SELECTOR = '.js-delete-user-modal-button';

export const MODAL_TEXTS_CONTAINER_SELECTOR = '#js-modal-texts';

export const MODAL_MANAGER_SELECTOR = '#js-delete-user-modal';
