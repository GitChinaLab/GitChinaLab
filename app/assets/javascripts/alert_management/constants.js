import { s__ } from '~/locale';

export const SEVERITY_LEVELS = {
  CRITICAL: s__('severity|Critical'),
  HIGH: s__('severity|High'),
  MEDIUM: s__('severity|Medium'),
  LOW: s__('severity|Low'),
  INFO: s__('severity|Info'),
  UNKNOWN: s__('severity|Unknown'),
};

export const ALERTS_STATUS_TABS = [
  {
    title: s__('AlertManagement|Open'),
    status: 'OPEN',
    filters: ['TRIGGERED', 'ACKNOWLEDGED'],
  },
  {
    title: s__('AlertManagement|Triggered'),
    status: 'TRIGGERED',
    filters: 'TRIGGERED',
  },
  {
    title: s__('AlertManagement|Acknowledged'),
    status: 'ACKNOWLEDGED',
    filters: 'ACKNOWLEDGED',
  },
  {
    title: s__('AlertManagement|Resolved'),
    status: 'RESOLVED',
    filters: 'RESOLVED',
  },
  {
    title: s__('AlertManagement|All alerts'),
    status: 'ALL',
    filters: ['TRIGGERED', 'ACKNOWLEDGED', 'RESOLVED'],
  },
];

/* eslint-disable @gitlab/require-i18n-strings */

/**
 * Tracks snowplow event when user views alerts list
 */
export const trackAlertListViewsOptions = {
  category: 'Alert Management',
  action: 'view_alerts_list',
};
