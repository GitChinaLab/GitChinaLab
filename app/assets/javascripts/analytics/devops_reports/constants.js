import { __ } from '~/locale';

export const INTRO_COOKIE_KEY = 'dev_ops_report_intro_callout_dismissed';

export const INTRO_BANNER_TITLE = __('Introducing Your DevOps Reports');

export const INTRO_BANNER_BODY = __(
  'Your DevOps Reports give an overview of how you are using GitLab from a feature perspective. Use them to view how you compare with other organizations, and how your teams compare against each other.',
);

export const INTRO_BANNER_ACTION_TEXT = __('Read more');
