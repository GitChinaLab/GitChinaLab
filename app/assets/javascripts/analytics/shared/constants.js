import { masks } from 'dateformat';

export const DATE_RANGE_LIMIT = 180;
export const OFFSET_DATE_BY_ONE = 1;
export const PROJECTS_PER_PAGE = 50;

const { isoDate, mediumDate } = masks;
export const dateFormats = {
  isoDate,
  defaultDate: mediumDate,
  defaultDateTime: 'mmm d, yyyy h:MMtt',
  month: 'mmmm',
};
