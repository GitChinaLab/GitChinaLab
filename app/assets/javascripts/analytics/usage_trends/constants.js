import { getDateInPast } from '~/lib/utils/datetime_utility';

export const TOTAL_DAYS_TO_SHOW = 365;
export const TODAY = new Date();
export const START_DATE = getDateInPast(TODAY, TOTAL_DAYS_TO_SHOW);
