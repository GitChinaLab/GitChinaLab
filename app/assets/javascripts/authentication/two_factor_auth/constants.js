export const COPY_BUTTON_ACTION = 'copy';
export const DOWNLOAD_BUTTON_ACTION = 'download';
export const PRINT_BUTTON_ACTION = 'print';

export const TRACKING_LABEL_PREFIX = '2fa_recovery_codes_';

export const RECOVERY_CODE_DOWNLOAD_FILENAME = 'gitlab-recovery-codes.txt';

export const SUCCESS_QUERY_PARAM = 'two_factor_auth_enabled_successfully';

export const COPY_KEYBOARD_SHORTCUT = 'mod+c';
