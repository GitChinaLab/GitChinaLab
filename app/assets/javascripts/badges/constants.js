export const GROUP_BADGE = 'group';
export const PROJECT_BADGE = 'project';
