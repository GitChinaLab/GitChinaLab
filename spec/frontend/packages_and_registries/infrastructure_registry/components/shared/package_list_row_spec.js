import { GlLink } from '@gitlab/ui';
import { shallowMountExtended } from 'helpers/vue_test_utils_helper';
import { createMockDirective, getBinding } from 'helpers/vue_mock_directive';

import PackagesListRow from '~/packages_and_registries/infrastructure_registry/shared/package_list_row.vue';
import PackagePath from '~/packages_and_registries/shared/components/package_path.vue';
import PackageTags from '~/packages_and_registries/shared/components/package_tags.vue';
import { PACKAGE_ERROR_STATUS } from '~/packages_and_registries/shared/constants';

import ListItem from '~/vue_shared/components/registry/list_item.vue';
import { packageList } from '../mock_data';

describe('packages_list_row', () => {
  let wrapper;
  let store;

  const [packageWithoutTags, packageWithTags] = packageList;

  const InfrastructureIconAndName = { name: 'InfrastructureIconAndName', template: '<div></div>' };

  const findPackageTags = () => wrapper.findComponent(PackageTags);
  const findPackagePath = () => wrapper.findComponent(PackagePath);
  const findDeleteButton = () => wrapper.findByTestId('action-delete');
  const findInfrastructureIconAndName = () => wrapper.findComponent(InfrastructureIconAndName);
  const findListItem = () => wrapper.findComponent(ListItem);
  const findPackageLink = () => wrapper.findComponent(GlLink);
  const findWarningIcon = () => wrapper.findByTestId('warning-icon');

  const mountComponent = ({
    isGroup = false,
    packageEntity = packageWithoutTags,
    showPackageType = true,
    disableDelete = false,
    provide,
  } = {}) => {
    wrapper = shallowMountExtended(PackagesListRow, {
      store,
      provide,
      stubs: {
        ListItem,
        InfrastructureIconAndName,
      },
      propsData: {
        packageLink: 'foo',
        packageEntity,
        isGroup,
        showPackageType,
        disableDelete,
      },
      directives: {
        GlTooltip: createMockDirective(),
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('renders', () => {
    mountComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('tags', () => {
    it('renders package tags when a package has tags', () => {
      mountComponent({ isGroup: false, packageEntity: packageWithTags });

      expect(findPackageTags().exists()).toBe(true);
    });

    it('does not render when there are no tags', () => {
      mountComponent();

      expect(findPackageTags().exists()).toBe(false);
    });
  });

  describe('when is is group', () => {
    it('has a package path component', () => {
      mountComponent({ isGroup: true });

      expect(findPackagePath().exists()).toBe(true);
      expect(findPackagePath().props()).toMatchObject({ path: 'foo/bar/baz' });
    });
  });

  describe('showPackageType', () => {
    it('shows the type when set', () => {
      mountComponent();

      expect(findInfrastructureIconAndName().exists()).toBe(true);
    });

    it('does not show the type when not set', () => {
      mountComponent({ showPackageType: false });

      expect(findInfrastructureIconAndName().exists()).toBe(false);
    });
  });

  describe('deleteAvailable', () => {
    it('does not show when not set', () => {
      mountComponent({ disableDelete: true });

      expect(findDeleteButton().exists()).toBe(false);
    });
  });

  describe('delete button', () => {
    it('exists and has the correct props', () => {
      mountComponent({ packageEntity: packageWithoutTags });

      expect(findDeleteButton().exists()).toBe(true);
      expect(findDeleteButton().attributes()).toMatchObject({
        icon: 'remove',
        category: 'secondary',
        variant: 'danger',
        title: 'Remove package',
      });
    });

    it('emits the packageToDelete event when the delete button is clicked', async () => {
      mountComponent({ packageEntity: packageWithoutTags });

      findDeleteButton().vm.$emit('click');

      await wrapper.vm.$nextTick();
      expect(wrapper.emitted('packageToDelete')).toBeTruthy();
      expect(wrapper.emitted('packageToDelete')[0]).toEqual([packageWithoutTags]);
    });
  });

  describe(`when the package is in ${PACKAGE_ERROR_STATUS} status`, () => {
    beforeEach(() => {
      mountComponent({ packageEntity: { ...packageWithoutTags, status: PACKAGE_ERROR_STATUS } });
    });

    it('list item has a disabled prop', () => {
      expect(findListItem().props('disabled')).toBe(true);
    });

    it('details link is disabled', () => {
      expect(findPackageLink().attributes('disabled')).toBe('true');
    });

    it('has a warning icon', () => {
      const icon = findWarningIcon();
      const tooltip = getBinding(icon.element, 'gl-tooltip');
      expect(icon.props('icon')).toBe('warning');
      expect(tooltip.value).toMatchObject({
        title: 'Invalid Package: failed metadata extraction',
      });
    });

    it('delete button is disabled', () => {
      expect(findDeleteButton().props('disabled')).toBe(true);
    });
  });
});
