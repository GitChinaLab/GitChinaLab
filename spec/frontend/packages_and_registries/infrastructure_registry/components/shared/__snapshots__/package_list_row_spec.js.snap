// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packages_list_row renders 1`] = `
<div
  class="gl-display-flex gl-flex-direction-column gl-border-b-solid gl-border-t-solid gl-border-t-1 gl-border-b-1 gl-border-t-transparent gl-border-b-gray-100"
  data-qa-selector="package_row"
>
  <div
    class="gl-display-flex gl-align-items-center gl-py-3"
  >
    <!---->
     
    <div
      class="gl-display-flex gl-xs-flex-direction-column gl-justify-content-space-between gl-align-items-stretch gl-flex-grow-1"
    >
      <div
        class="gl-display-flex gl-flex-direction-column gl-xs-mb-3 gl-min-w-0 gl-flex-grow-1"
      >
        <div
          class="gl-display-flex gl-align-items-center gl-text-body gl-font-weight-bold gl-min-h-6 gl-min-w-0"
        >
          <div
            class="gl-display-flex gl-align-items-center gl-mr-3 gl-min-w-0"
          >
            <gl-link-stub
              class="gl-text-body gl-min-w-0"
              data-qa-selector="package_link"
              href="foo"
            >
              <gl-truncate-stub
                position="end"
                text="Test package"
              />
            </gl-link-stub>
             
            <!---->
             
            <!---->
          </div>
           
          <!---->
        </div>
         
        <div
          class="gl-display-flex gl-align-items-center gl-text-gray-500 gl-min-h-6 gl-min-w-0 gl-flex-grow-1"
        >
          <div
            class="gl-display-flex"
          >
            <span>
              1.0.0
            </span>
             
            <!---->
             
            <div />
             
            <package-path-stub
              path="foo/bar/baz"
            />
          </div>
        </div>
      </div>
       
      <div
        class="gl-display-flex gl-flex-direction-column gl-sm-align-items-flex-end gl-justify-content-space-between gl-text-gray-500 gl-flex-shrink-0"
      >
        <div
          class="gl-display-flex gl-align-items-center gl-sm-text-body gl-sm-font-weight-bold gl-min-h-6"
        >
          <publish-method-stub
            packageentity="[object Object]"
          />
        </div>
         
        <div
          class="gl-display-flex gl-align-items-center gl-min-h-6"
        >
          <span>
            <gl-sprintf-stub
              message="Created %{timestamp}"
            />
          </span>
        </div>
      </div>
    </div>
     
    <div
      class="gl-w-9 gl-display-flex gl-justify-content-end gl-pr-1"
    >
      <gl-button-stub
        aria-label="Remove package"
        buttontextclasses=""
        category="secondary"
        data-testid="action-delete"
        icon="remove"
        size="medium"
        title="Remove package"
        variant="danger"
      />
    </div>
  </div>
   
  <div
    class="gl-display-flex"
  >
    <div
      class="gl-w-7"
    />
     
    <!---->
     
    <div
      class="gl-w-9"
    />
  </div>
</div>
`;
